frontend/src/lib/components/Users/CreateUser.svelte

```diff
@@ -0,0 +1,98 @@
+<script lang="ts">
+  import PasswordStrengthMeter from '$lib/components/PasswordStrengthMeter.svelte';
+  import { SubmitButton, FormError, Input, Form, ProtectedForm, isEmail, lexSuperForm, passwordFormRules, DisplayLanguageSelect } from '$lib/forms';
+  import t, { getLanguageCodeFromNavigator, locale } from '$lib/i18n';
+  import { type RegisterResponse } from '$lib/user';
+  import { getSearchParamValues } from '$lib/util/query-params';
+  import { createEventDispatcher, onMount } from 'svelte';
+  import { usernameRe } from '$lib/user';
+  import { z } from 'zod';
+
+  export let allowUsernames = false;
+  export let skipTurnstile = false;
+  export let submitButtonText = $t('register.button_register');
+  export let handleSubmit: (password: string, passwordStrength: number, name: string, email: string, locale: string, turnstileToken: string) => Promise<RegisterResponse>;
+
+  const dispatch = createEventDispatcher();
+
+  type RegisterPageQueryParams = {
+    name: string;
+    email: string;
+  };
+  let turnstileToken = '';
+  // $locale is the locale that our i18n is using for them (i.e. the best available option we have for them)
+  // getLanguageCodeFromNavigator() gives us the language/locale they probably actually want. Maybe we'll support it in the future.
+  const userLocale = getLanguageCodeFromNavigator() ?? $locale;
+  const formSchema = z.object({
+    name: z.string().trim().min(1, $t('register.name_missing')),
+    email: z.string().trim()
+      .min(1, $t('project_page.add_user.empty_user_field'))
+      .refine((value) => isEmail(value) || (allowUsernames && usernameRe.test(value)), { message: $t('register.invalid_input') }),
+    password: passwordFormRules($t),
+    score: z.number(),
+    locale: z.string().trim().min(2).default(userLocale),
+  });
+
+  let { form, errors, message, enhance, submitting } = lexSuperForm(formSchema, async () => {
+    const { user, error } = await handleSubmit($form.password, $form.score, $form.name, $form.email, $form.locale, turnstileToken);
+    if (error) {
+      if (error.turnstile) {
+        $message = $t('turnstile.invalid');
+      }
+      if (error.accountExists) {
+        $errors.email = [$t('register.account_exists')];
+      }
+      if (error.invalidInput) {
+        $errors.email = [$t('register.invalid_input')];
+      }
+      return;
+    }
+    if (user) {
+      dispatch('submitted');
+      return;
+    }
+    throw new Error('Unknown error, no error from server, but also no user.');
+  });
+  onMount(() => { // query params not available during SSR
+    const urlValues = getSearchParamValues<RegisterPageQueryParams>();
+    form.update((form) => {
+      if (urlValues.name) form.name = urlValues.name;
+      if (urlValues.email) form.email = urlValues.email;
+      return form;
+    }, { taint: true });
+  });
+</script>
+
+<svelte:component this={skipTurnstile ? Form : ProtectedForm} {enhance} bind:turnstileToken>
```

Comment:
I was actually expecting a compile error because `turnstileToken` isn't a property on Form. We could, in theory, solve this by making a component called something like `MaybeProtectedForm` which looks like this:

```html
{#if skipTurnstile}
<Form {enhance}><slot /></Form>
{:else}
<ProtectedForm {enhance} bind:turnstileToken><slot /></ProtectedForm>
{/if}
```

But that gets a little ridiculous. The other alternative (besides leaving it as-is) would be to add `export let turnstileToken = ''` to the Form component, which I also don't like since it's not actually used in Form and would only be there to make this one line type-check (so to speak).

I'll open an issue to track this; I agree that I don't like letting warnings pile up ignored, because you soon start ignoring real problems, but I don't see a *good* solution right now.

Reply:
Actually, one good solution that just occurred to me would be to just continue using ProtectedForm even though we do trust our admins, because it doesn't cost them any effort. @myieye, you were the one who suggested removing the turnstile token for the admin-only create user modal. Would you be okay with reverting that change and leaving the turnstile token in the form even for admins? I don't think there's much drawback to doing that.

Reply:
Did a bit of digging into the Svelte 4 source and found that that warning is not suppressible with `<!-- svelte-ignore -->`, but that it only shows up in dev mode. So at this point, I'm thinking that perhaps that MaybeProtectedForm component is the way to go. It feels silly, but it's not all that large a component and it'll let us get rid of that error. Running out of time to do that today, though, so I'll have to do it tomorrow.

Reply:
For the record, it was @hahn-kev that [brought up](https://github.com/sillsdev/languageforge-lexbox/pull/736#pullrequestreview-2055873940) removing turnstile, but I do agree.
I'd approve any of these options:
- Throwing away ProtectedForm and making turnstile an optional feature of Form
- Introducing a new component to pick between the two form components
- Living with the warning

Reply:
I like the first 2 options. But let's move this into it's own issue since it's not a simple solve on this already large PR